### PR TITLE
Allow easily switching colors for playing multi-handed solo.

### PR DESCRIPTION
**Pull request by tomprince**

### CODE REVIEW

File: global-script.lua [hunk 1]
```diff
@@ -3079,8 +3087,30 @@ function tCompare(t1,t2)
...
+    Wait.frames(function ()
+        Player[color].lookAt({
+            position = getObjectFromGUID(playerCounters[color]).getPosition() - Vector(0,0,15),
+            pitch = 70,
+            distance = 30,
+        })
+    end, 2)
+end
+function onPlayerChangeColor(color)
+    for color, counterGUID in pairs(playerCounters) do
```
Comment (iakona) [inline global-script.lua:3107]: you have two variables in this function called `color`

File: global-script.lua [hunk 2]
```diff
@@ -3079,8 +3087,30 @@ function tCompare(t1,t2)
...
+        counter = getObjectFromGUID(counterGUID)
+        if Player[color].seated then
```
Comment (iakona) [inline global-script.lua:3109]: you could also check to make sure that the seat has a spirit picked something along the lines of `#getObjectFromGUID(PlayerBags[color]).getObjects() == 0`

File: global-script.lua [hunk 3]
```diff
@@ -3079,8 +3087,30 @@ function tCompare(t1,t2)
...
+    player.changeColor(color)
+    Wait.frames(function ()
+        Player[color].lookAt({
+            position = getObjectFromGUID(playerCounters[color]).getPosition() - Vector(0,0,15),
+            pitch = 70,
+            distance = 30,
+        })
+    end, 2)
+end
+function onPlayerChangeColor(color)
```
Comment (iakona) [inline global-script.lua:3106]: you probably want to extract this code out into a separate function and then call that here for `onPlayerChangeColor` as well as `onLoad` and when the game starts

File: global-script.lua [hunk 4]
```diff
@@ -3079,8 +3087,30 @@ function tCompare(t1,t2)
...
+    for color, counterGUID in pairs(playerCounters) do
+        counter = getObjectFromGUID(counterGUID)
```
Comment (iakona) [inline global-script.lua:3108]: during onLoad you could lookup all of the counters at once so you don't have to look them up each time the button is pressed

File: global-script.lua (hunk 2 above)
Comment (iakona) [reply global-script.lua:3109]: so on second thought maybe you have the buttons available if the game hasn't started, but then once the game has started all buttons for non seats without spirits go away?

Comment (tomprince) [reply global-script.lua:3243]: I'd rather not be that prescriptive. I'm fairly certain that I've selected spirits after start the game, at times.

**Comment by tomprince** [comment]
This change has interactions with the behavior in #8, that I've not looked at.

File: global-script.lua (hunk 2 above)
Comment (iakona) [reply global-script.lua:3243]: alright so here's my thinking. before the game has started the button will show for all empty seats. but once the game has started the button only shows for empty seats which contain a spirit

Comment (tomprince) [reply global-script.lua:3243]: My point is that I sometimes start the game before picking spirits, so I want to switch to seats without a spirit, to be able to pick them. I definitely understand the reasoning for wanting to remove them, but I think being less prescriptive makes sense here.

Comment (iakona) [reply global-script.lua:3243]: you can still switch seats the old fashioned way, just not a fan of cluttering up that space with buttons forever sticking around. just made a change last night to remove all the text and ready tokens until after a spirit has been picked

Comment (tomprince) [reply global-script.lua:3243]: I think the flexibility is worth a (tiny) bit of clutter somewhere that most people aren't going to be paying attention to anyway (I similarly am not convinced removing the energy tracker makes sense). For example, I'm not sure why someone who wants to play against a randomized adversary/scenario but choose spirits afterward should be punished by having to use a slightly more cumbersome way to switch between colors to pick spirits.

In any case, I will definitely patch the current behavior in locally, though I'd prefer not to have to do that.

**Comment by iakona** [comment]
So i merged in dewin's changes earlier today which contains most of these changes. iirc the main thing missing is the camera focus changes

**Pull request closed without merging**